### DISCUSSION TITLE
Exception when using date field or range of dates.

### DISCUSSION THREAD

**Discussion by SKUDSIEH** [fuzzzerd/fmdata#253] [Q&A] [answered]
Hello,
  Every time i try to use FindAsync with a date i get an "Exception: Could not read response from Data API." exception. If i use the DateOnly variable type i get an "Exception: Date value does not meet validation entry options" error and if i use DateTime i get an  "Exception: Parameter value is invalid" error. My FindAsync works fine with other types of fields like string or int.

public Model model = new Model() { dateVariable = new DateOnly (2022,10,04) }
var result = await FileMakerRestClient.FindAsync(model);

I would also like to know how to query for a range of dates like we do in filemaker. ( 2022-10-01... 2022-10-30)

**Comment by fuzzzerd** [comment, marked as answer]
I have solved this particular issue as follows.

Create two models.

The main model that you will use:

```cs
[DataContract("YourLayout")]
public class Model
{
    [DataMember]
    // this can be helpful if your field in FileMaker is a FileMaker Date field vs a Timestamp.
    [JsonConverter(typeof(FileMakerDateFormatConverter))]
    public virtual DateTime? DateField { get; set; }
}
```

A model you will use to search with:

```cs
[DataContract("YourLayout")]
public class QueryModel
{
    [DataMember(Name="DateField")]
    public string DateFieldString { get; set; }
}
```
That allows you to use a query model with a string that actually is Serialized into the `DateField` that is in your database/layout but using a .NET String which allows for the FileMaker specific find variation with the `..` operator.

```cs
QueryModel query = new QueryModel { DateFieldString = "10/01/2022..10/04/2022" };
QueryModel result = await client.FindAsync(query);
// this is where I'd use automapper to make this more consistent
Model model = new Model { DateField = result.DateField };
```

Hope that is helpful.
